udf-compiler/src/main/scala/com/nvidia/spark/udf/Plugin.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.udf
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.SparkSessionExtensions
+import org.apache.spark.sql.catalyst.expressions.{Alias, Expression, NamedExpression, ScalaUDF}
+import org.apache.spark.sql.catalyst.plans.logical.{LogicalPlan, Project}
+import org.apache.spark.sql.catalyst.rules.Rule
+
+import com.nvidia.spark.rapids.RapidsConf
+
+class Plugin extends Function1[SparkSessionExtensions, Unit] with Logging {
+  override def apply(extensions: SparkSessionExtensions): Unit = {
+    logWarning("Installing extensions to enable rapids UDF compiler support." +
+      s" To disable rapids UDF compiler support set [a setting] to false")
+    extensions.injectResolutionRule(_ => LogicalPlanRules())
+  }
+}
+
+//TODO is this it? Do I need more things here (alessandro)
```

Comment:
Nope it isn't. The main comment was made because I think other `LogicanPlan` nodes may contain UDFs, so I think we need to address those. That said, this plugin could at first claim "replaces udfs in `Project`s", as that's what it does right now.

Reply:
TODO removed.